pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -212,56 +280,59 @@ func (r *Runtime) RunGadget(gadgetCtx runtime.GadgetContext) (runtime.CombinedGa
 	results := make(runtime.CombinedGadgetResult)
 	var resultsLock sync.Mutex
 
-	allParams := make(map[string]string)
-	gadgets.ParamsToMap(
-		allParams,
-		gadgetCtx.GadgetParams(),
-		gadgetCtx.RuntimeParams(),
-		gadgetCtx.OperatorsParamCollection(),
-	)
-
-	gadgetCtx.Logger().Debugf("Params")
-	for k, v := range allParams {
-		gadgetCtx.Logger().Debugf("- %s: %q", k, v)
-	}
-
 	wg := sync.WaitGroup{}
-	for _, pod := range pods {
+	for _, t := range targets {
 		wg.Add(1)
-		go func(pod gadgetPod) {
-			gadgetCtx.Logger().Debugf("running gadget on node %q", pod.node)
-			res, err := r.runGadget(gadgetCtx, pod, allParams)
+		go func(target target) {
+			gadgetCtx.Logger().Debugf("running gadget on node %q", target.node)
+			res, err := r.runGadget(gadgetCtx, target, paramMap)
 			resultsLock.Lock()
-			results[pod.node] = &runtime.GadgetResult{
+			results[target.node] = &runtime.GadgetResult{
 				Payload: res,
 				Error:   err,
 			}
 			resultsLock.Unlock()
 			wg.Done()
-		}(pod)
+		}(t)
 	}
 
 	wg.Wait()
 	return results, results.Err()
 }
 
-func (r *Runtime) runGadget(gadgetCtx runtime.GadgetContext, pod gadgetPod, allParams map[string]string) ([]byte, error) {
+func (r *Runtime) dialContext(dialCtx context.Context, target target) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	}
+
+	// If we're in Kubernetes connection mode, we need a custom dialer
+	if r.connectionMode == ConnectionModeKubernetesProxy {
+		opts = append(opts, grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+			return NewK8SExecConn(ctx, target, time.Second*ConnectTimeout)
+		}))
+	}
+
+	conn, err := grpc.DialContext(dialCtx, "passthrough:///"+target.name, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("dialing node %q: %w", target.node, err)
```

Comment:
nit: this message is misleading when running without kubernetes as node is empty. 

```
WARN[0030] could not load gadget info from remote: dialing random target: dialing node "": context deadline exceeded
```